core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -73,8 +71,7 @@ void DeepCopyAsyncCuda(const Cuda &instance, void *dst, const void *src,
 void DeepCopyAsyncCuda(void *dst, const void *src, size_t n) {
   cudaStream_t s = cuda_get_deep_copy_stream();
   KOKKOS_IMPL_CUDA_SAFE_CALL(
-      (CudaInternal::singleton().cuda_memcpy_async_wrapper(
-          dst, src, n, cudaMemcpyDefault, s)));
+      cudaMemcpyAsync(dst, src, n, cudaMemcpyDefault, s));
```

Comment:
How do we ensure that we are using the (default) device the stream is associated with?

Reply:
Yeah, we need to set device. I'll add it.

Reply:
This is a special case. The wrappers which took a optional stream were previously incorrect (but never used incorrectly) since they didn't query the stream for the device  associated to set it correctly, they just used the internal device. I chose to still abandon the optional string wrapper and call CUDA API. I set the device to the singleton one since that is what we used to create this stream.

Reply:
Do we still want these changes then?

Reply:
I'd say revert this

Reply:
Fine